core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala

```diff
@@ -207,6 +210,15 @@ class HadoopRDD[K, V](
       } else {
         allInputSplits
       }
+      if (inputSplits.length == 1 && inputSplits(0).isInstanceOf[FileSplit]) {
+        val fileSplit = inputSplits(0).asInstanceOf[FileSplit]
+        val path = fileSplit.getPath
+        if (Utils.isFileSplittable(path, codecFactory)
```

Comment:
do we really need to know it's splittable or not? If Spark is scanning files with a single giant partition, it's going to be very slow.

Reply:
@cloud-fan Yes. But we'd better tell user why it only generate only one partition. So I prefer:
* If the file is unsplittable, then in log tell user the file is unsplittable (and include unsplittable reason)
* If the file is splittable, then in log tell user we can increase parallelism by setting the argument `minPartitions` in method `sc.textFile`.
What do you think ?

Reply:
@cloud-fan Any thoughts ?

Reply:
SGTM, let's include the reason in the message.